receiver/googlecloudmonitoringreceiver/receiver.go

```diff
@@ -5,51 +5,331 @@ package googlecloudmonitoringreceiver // import "github.com/open-telemetry/opent
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/iterator"
+	"google.golang.org/api/option"
+	"google.golang.org/genproto/googleapis/api/metric"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/googlecloudmonitoringreceiver/internal"
 )
 
 type monitoringReceiver struct {
-	config *Config
-	logger *zap.Logger
-	cancel context.CancelFunc
+	config            *Config
+	logger            *zap.Logger
+	client            *monitoring.MetricClient
+	metricsBuilder    *internal.MetricsBuilder
+	mutex             sync.RWMutex
+	metricDescriptors map[string]*metric.MetricDescriptor
 }
 
 func newGoogleCloudMonitoringReceiver(cfg *Config, logger *zap.Logger) *monitoringReceiver {
 	return &monitoringReceiver{
-		config: cfg,
-		logger: logger,
+		config:         cfg,
+		logger:         logger,
+		metricsBuilder: internal.NewMetricsBuilder(logger),
 	}
 }
 
-func (m *monitoringReceiver) Scrape(ctx context.Context) (pmetric.Metrics, error) {
-	// Dummy use to fix lint errors
-	ctx.Deadline()
+func (mr *monitoringReceiver) Start(ctx context.Context, _ component.Host) error {
+	mr.metricDescriptors = make(map[string]*metric.MetricDescriptor)
+
+	// Lock to ensure thread-safe access to mr.client
+	mr.mutex.Lock()
+	defer mr.mutex.Unlock()
+
+	// Skip client initialization if already initialized
+	if mr.client == nil {
+		if err := mr.initializeClient(ctx); err != nil {
+			return err
+		}
+		mr.logger.Info("Monitoring client successfully created.")
+	}
+
+	// Initialize metric descriptors, even if the client was previously initialized
+	if len(mr.metricDescriptors) == 0 {
+		if err := mr.initializeMetricDescriptors(ctx); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (mr *monitoringReceiver) Shutdown(context.Context) error {
+	mr.mutex.Lock()
+	defer mr.mutex.Unlock()
+
+	var err error
+	if mr.client != nil {
+		err = mr.client.Close()
+	}
+	return err
+}
+
+func (mr *monitoringReceiver) Scrape(ctx context.Context) (pmetric.Metrics, error) {
+	var (
+		gInternal    time.Duration
+		gDelay       time.Duration
+		calStartTime time.Time
+		calEndTime   time.Time
+		filterQuery  string
+		gErr         error
+	)
+
 	metrics := pmetric.NewMetrics()
-	m.logger.Debug("Scrape metrics ")
 
-	return metrics, nil
+	// Iterate over each metric in the configuration to calculate start/end times and construct the filter query.
+	for _, metric := range mr.config.MetricsList {
+		// Acquire read lock to safely read metricDescriptors
+		mr.mutex.RLock()
+		metricDesc, exists := mr.metricDescriptors[metric.MetricName]
+		mr.mutex.RUnlock()
+		if !exists {
+			mr.logger.Warn("Metric descriptor not found", zap.String("metric_name", metric.MetricName))
+			continue
+		}
+
+		// Set interval and delay times, using defaults if not provided
+		gInternal = mr.config.CollectionInterval
+		if gInternal <= 0 {
+			gInternal = defaultCollectionInterval
+		}
+
+		gDelay = metric.FetchDelay
```

Comment:
I learned from another Googler that the fetch delay is actually provided by the metric descriptor: https://pkg.go.dev/google.golang.org/genproto/googleapis/api/metric#MetricDescriptor_MetricDescriptorMetadata

You can get it with
```golang
gDelay = metricDesc.GetMetadata().GetIngestDelay().AsDuration()
```

You can actually remove this from the configuration as well.

Reply:
Remove the **fetch_delay** from the configuration file and instead retrieve it from the metric descriptor.